libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/daemon/MessageBusListenerActor.scala

```diff
@@ -32,9 +36,11 @@ class MessageBusListenerActor[M](source: Source[M, NotUsed])(implicit messageLik
         log.warning("Listener already subscribed. Ignoring Subscribe message")
       case Failure(ex) =>
         log.error(ex, "Source/Listener died, subscribing again")
+        monitor.onError(ex)
         trySubscribeDelayed()
         context become idle
       case Done =>
+        monitor.onFinished
         log.info("Source finished, subscribing again")
         trySubscribeDelayed()
```

Comment:
I pushed a commit to do that but I am not entirely sure we should do that. We need to trust that that akka-stream-kafka will never terminate early, which is what it's supposed to do, but problems happen. In case that happens then our actor will just die and will not be restarted by anything.